src/main/java/pacApp/pacController/RentalController.java

```diff
@@ -2,86 +2,297 @@
 import pacApp.pacData.RentalRepository;
+import pacApp.pacData.UserRepository;
+import pacApp.pacException.AuthenticationForbiddenException;
 import pacApp.pacException.RentalNotFoundException;
 import pacApp.pacModel.Car;
 import pacApp.pacModel.Rental;
+import pacApp.pacModel.User;
+import pacApp.pacModel.request.Booking;
+import pacApp.pacModel.response.GenericResponse;
+import pacApp.pacSecurity.JwtAuthenticatedProfile;
 
-import java.util.ArrayList;
+import java.awt.print.Book;
+import java.sql.Timestamp;
+import java.time.Instant;
 import java.util.List;
+import java.util.Optional;
+import java.util.Vector;
 
 @RestController
 @Api(value = "RentalController", description = "Operations pertaining to Rentals in Car Rental System")
 public class RentalController {
 
     private final RentalRepository repository;
+    private final UserRepository userRepository;
+    private final CarRepository carRepository;
     private static final Logger log = LoggerFactory.getLogger(RentalController.class);
 
-    public RentalController(RentalRepository repository){
+    public RentalController(RentalRepository repository, UserRepository userRepository, CarRepository carRepository){
         this.repository = repository;
+        this.userRepository = userRepository;
+        this.carRepository = carRepository;
     }
 
     @GetMapping("/rental")
-    public List<Rental> getAllRentals(){
+    public List<Booking> getAllRentals(){
         Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+
+        if (!(auth instanceof JwtAuthenticatedProfile)) {
+            throw new AuthenticationForbiddenException("authentication failure");
+        }
+
         log.info(auth.toString());
         log.info(Boolean.toString(auth.isAuthenticated()));
         log.info(auth.getName());
         log.info(auth.getPrincipal().toString());
 
-        return new ArrayList<>();
-        //return this.repository.findAll();
+        JwtAuthenticatedProfile authenticatedProfile = (JwtAuthenticatedProfile) auth;
+
+        String userEmail = authenticatedProfile.getName();
+        Optional<User> optUser = this.userRepository.findOneByEmail(userEmail);
+
+        if (!optUser.isPresent()){
+            throw new AuthenticationForbiddenException("authentication failure");
+        }
+
+        User user = optUser.get();
+        List<Rental> rentals = this.repository.findByUser(user);
+        List<Booking> bookings = new Vector<Booking>();
+
+        for(Rental rental : rentals) {
+            bookings.add(this.convertRentalToBooking(rental));
+        }
+
+        return bookings;
     }
 
     @GetMapping("/rental/{id}")
-    public Rental getRental(@PathVariable Long id){
-        return this.repository.findById(id)
-                .orElseThrow(() -> new RentalNotFoundException(id));
-    }
+    public ResponseEntity getRental(@PathVariable Long id){
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
If this could be influenced by the user, please check for malicious code before processing it